lib/terminal-view.js

```diff
@@ -210,30 +212,30 @@ class TerminalView extends View {
       return
     }
 
-    var height = (this.outerHeight() + this.offset().top) - pageY;
-    if ((height < 100) && (height < this.height())) { return }
+    var availableHeight = this.height() + this.offset().top;
+    var proposedHeight  = availableHeight - pageY;
 
-    this.height(height)
+    var tooLarge = pageY < verticalLimit;
+    var tooSmall = proposedHeight < verticalLimit;
+
+    if (tooLarge) { proposedHeight = availableHeight - verticalLimit }
+    if (tooSmall) { proposedHeight = verticalLimit }
+
+    this.height(proposedHeight)
     this.fit()
   }
 
   fit() {
-    this.emulator.fit();
+    // Two calls are necessary to properly fit after the font-size has changed
+    this.emulator.fit()
+    this.emulator.fit()
```

Comment:
@ktravers if you're interested, this is the magic that seems to have guaranteed, finally (the last attempt actually could still break), that the emulator will fit inside its panel 🙄 

I _think_ what's happening is that the first call to `emulator.fit()` is pushing xterm to adjust to font-size changes we've made, and the second call is then taking that and actually fitting the emulator in its container. That's just a hunch though. Whatever it is, it works 🤷‍♂️